spec/models/borrowed_request_spec.rb

```diff
@@ -0,0 +1,82 @@
+require "rails_helper"
+
+RSpec.describe BorrowedRequest, type: :model do
+  describe "associations" do
+    it {should belong_to(:user)}
+  end
+
+  describe "scopes" do
+    let!(:user) {FactoryBot.create :user}
+    let!(:request_1) {FactoryBot.create :borrowed_request, user_id: user.id, status: 1}
+    let!(:request_2) {FactoryBot.create :borrowed_request, user_id: user.id}
+
+    context "order by status" do
+      it "should be order by status" do
+        expect(BorrowedRequest.order_status).to eq [request_2, request_1]
+      end
+    end
+  end
+
+  describe "delegates" do
+    let!(:user) {FactoryBot.create :user}
+    let!(:request) {FactoryBot.create :borrowed_request, user_id: user.id}
+
+    context "user's name" do
+      it "should return user's name" do
+        expect(request.user_name).to eq request.user.name
+      end
+    end
+  end
+
+  describe "enums" do
+    let!(:user) {FactoryBot.create :user}
+    let!(:request) {FactoryBot.create :borrowed_request, user_id: user.id}
+
+    context "change status" do
+      it "should requested" do
+        request.update status: 0
+        expect(request.requested?).to eq true
+      end
+
+      it "should borrowing" do
+        request.update status: 1
+        expect(request.borrowing?).to eq true
+      end
+
+      it "should returned" do
+        request.update status: 2
+        expect(request.returned?).to eq true
+      end
+
+      it "should rejected" do
+        request.update status: 3
+        expect(request.rejected?).to eq true
+      end
```

Comment:
tesst enum chỉ cần 
`it { should define_enum_for(:status).
            with(:.... tự điền giá trị em định nghĩa là dc) }`